backend/api/EventHandlers/InspectionFindingEventHandler.cs

```diff
@@ -77,19 +81,32 @@ private async Task<List<Finding>> GenerateFindingsList(List<InspectionFinding> i
                 var missionRun = await InspectionFindingService.GetMissionRunByIsarStepId(inspectionFinding);
                 var task = await InspectionFindingService.GetMissionTaskByIsarStepId(inspectionFinding);
 
-                if (task?.TagId != null && missionRun?.Area?.Plant?.Name != null && missionRun?.Area?.Name != null)
+                if (task != null && missionRun != null && missionRun?.Area != null)
                 {
                     var finding = new Finding(
-                        task.TagId,
-                        missionRun.Area.Plant.Name,
-                        missionRun.Area.Name,
+                        task.TagId ?? "NA",
+                        missionRun.Area.Plant.Name ?? "NA",
+                        missionRun.Area.Name ?? "NA",
                         inspectionFinding.Finding,
                         inspectionFinding.InspectionDate,
-                        missionRun.Robot.Name
+                        missionRun.Robot.Name ?? "NA"
                     );
 
                     findingsList.Add(finding);
                 }
+                else if (task != null && missionRun?.Area == null && missionRun != null)
```

Comment:
With the suggestions above I think you can drop this else

Reply:
Agreed, done.